packages/framework/ai-collab/src/explicit-strategy/idGenerator.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert, oob } from "@fluidframework/core-utils/internal";
+import { Tree, NodeKind } from "@fluidframework/tree/internal";
+import type {
+	TreeNode,
+	ImplicitFieldSchema,
+	TreeArrayNode,
+	TreeFieldFromImplicitField,
+} from "@fluidframework/tree/internal";
+
+/**
+ * Given a tree node, generates a set of LLM -friendly, unique IDs for each node in a given Shared Tree.
+ * @remarks simple id's are important for the LLM and this library to create and distinguish between different types certain TreeEdits
```

Comment:
Nit (sorry, I'm a very docs-focused person, so I can be very picky about our comments 😅)
```suggestion
 * @remarks The ability to uniquely and stably in the tree is important for the LLM and this library to create and distinguish between different types certain {@link TreeEdit}s.
```

Would probably also benefit from an example scenario where this is important. Like allowing the LLM to move a node in an array to be adjacent to some other node, while other changes are being made to the tree.

Reply:
Comment updated in commit b700ced170a403db24c887b25ce4917392948b3c , still pending adding an example scenario